app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -333,12 +334,30 @@ def redirect_url
       if !mobile? && TwoFactorAuthentication::PivCacPolicy.new(current_user).enabled?
         login_two_factor_piv_cac_url
       elsif TwoFactorAuthentication::WebauthnPolicy.new(current_user).enabled?
-        login_two_factor_webauthn_url(webauthn_params)
+        webauthn_url
       elsif TwoFactorAuthentication::AuthAppPolicy.new(current_user).enabled?
         login_two_factor_authenticator_url
       end
     end
 
+    def webauthn_url
+      if webauthn_params[:platform] == true
+        if !device_supports_webauthn_platform?
+          return login_two_factor_options_url
+        end
+      end
+      login_two_factor_webauthn_url(webauthn_params)
```

Comment:
We could flatten the two `if`s here, also possibly arrange as an `if`/`else` vs. an early return.

```suggestion
      if webauthn_params[:platform] == true && !device_supports_webauthn_platform?
        login_two_factor_options_url
      else
        login_two_factor_webauthn_url(webauthn_params)
      end
```